### PR TITLE
generate files

### PR DESCRIPTION
**Pull request by elie222**

### CODE REVIEW

File: aleph0/index.ts [hunk 1]
```diff
@@ -17,23 +23,34 @@ async function generate(text: string) {
...
-    await createChangesArray(relevantSnippet, projectSturcutre, "nextjs13"),
+    await createChangesArray(relevantSnippet, projectStructure, "nextjs13"),
     "Find what files are relevant for these snippets in this project.",
     "gpt-4"
   );
   if (!changesRaw)
     throw new Error(`AI returned a bad changes array: ${relevantSnippet}`);
 
-  const changes = JSON.parse(changesRaw);
+  const changesSchema = z.array(
```
Comment (agamm) [inline aleph0/index.ts:36]: I love this sanity check.

File: aleph0/lib/file.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export function createFile(filePath: string, content: string) {
+  const resolvedPath = path.resolve(filePath);
+  const dirName = path.dirname(resolvedPath);
+
+  // Check if the directory exists
+  if (!fs.existsSync(dirName)) {
+    // If not, create the directory and any nested directories that might be needed
+    fs.mkdirSync(dirName, { recursive: true });
+    // consola.success(`Directory ${dirName} created.`);
```
Comment (agamm) [inline aleph0/lib/file.ts:19]: Let's remove unnecessary comments?

File: aleph0/lib/file.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export function replaceFile(filePath: string, content: string, log = true) {
+  const resolvedPath = path.resolve(filePath);
+  const dirName = path.dirname(resolvedPath);
+
+  // Check if the directory exists
+  if (!fs.existsSync(dirName)) {
+    // If not, create the directory and any nested directories that might be needed
+    fs.mkdirSync(dirName, { recursive: true });
+    // consola.success(`Directory ${dirName} created.`);
```
Comment (agamm) [inline aleph0/lib/file.ts:34]: Here too.

**Review by agamm — APPROVED** [review]
LGTM

File: aleph0/lib/file.ts (hunk 2 above)
Comment (elie222) [reply aleph0/lib/file.ts:19]: sure. i just copy and pasted this from https://github.com/nicoalbanese/kirimase/blob/master/src/utils.ts

**Pull request merged**